src/vse_sync_pp/analyzers/pmc.py

```diff
@@ -0,0 +1,134 @@
+### SPDX-License-Identifier: GPL-2.0-or-later
+
+"""Analyze PMC log messages"""
+
+from .analyzer import Analyzer
+import copy
+
+STATE_FREERUN = 248
+STATE_LOCKED = 6
+STATE_HOLDOVER_IN_SPEC = 7
+STATE_HOLDOVER_OUT_OF_SPEC1 = 140
+STATE_HOLDOVER_OUT_OF_SPEC2 = 150
+STATE_HOLDOVER_OUT_OF_SPEC3 = 160
+
+STATE_NAMES = {
+   STATE_FREERUN: "FREERUN",
+   STATE_LOCKED: "LOCKED",
+   STATE_HOLDOVER_IN_SPEC: "HOLDOVER_IN_SPEC",
+   STATE_HOLDOVER_OUT_OF_SPEC1: "HOLDOVER_OUT_SPEC1",
+   STATE_HOLDOVER_OUT_OF_SPEC2: "HOLDOVER_OUT_SPEC2",
+   STATE_HOLDOVER_OUT_OF_SPEC3: "HOLDOVER_OUT_SPEC3",
+}
+
+STATE_TRANSITION = {
+    STATE_FREERUN: [STATE_FREERUN, STATE_LOCKED],
+    STATE_LOCKED: [STATE_LOCKED, STATE_HOLDOVER_IN_SPEC],
+    STATE_HOLDOVER_IN_SPEC: [STATE_LOCKED, 
+                             STATE_HOLDOVER_IN_SPEC, 
+                             STATE_HOLDOVER_OUT_OF_SPEC1, 
+                             STATE_HOLDOVER_OUT_OF_SPEC2, 
+                             STATE_HOLDOVER_OUT_OF_SPEC3],
+    STATE_HOLDOVER_OUT_OF_SPEC1: [STATE_LOCKED, 
+                                  STATE_HOLDOVER_OUT_OF_SPEC1, 
+                                  STATE_HOLDOVER_OUT_OF_SPEC2, 
+                                  STATE_HOLDOVER_OUT_OF_SPEC3],
+    STATE_HOLDOVER_OUT_OF_SPEC2: [STATE_LOCKED, 
+                                  STATE_HOLDOVER_OUT_OF_SPEC1, 
+                                  STATE_HOLDOVER_OUT_OF_SPEC2, 
+                                  STATE_HOLDOVER_OUT_OF_SPEC3],
+    STATE_HOLDOVER_OUT_OF_SPEC3: [STATE_LOCKED, 
+                                  STATE_HOLDOVER_OUT_OF_SPEC1, 
+                                  STATE_HOLDOVER_OUT_OF_SPEC2, 
+                                  STATE_HOLDOVER_OUT_OF_SPEC3],
+}
+
+BASE_CLOCK_CLASS_COUNT = {
+    "count": 0,
+    "transitions": {
+        STATE_FREERUN: 0,
+        STATE_LOCKED: 0,
+        STATE_HOLDOVER_IN_SPEC: 0,
+        STATE_HOLDOVER_OUT_OF_SPEC1: 0,
+        STATE_HOLDOVER_OUT_OF_SPEC2: 0,
+        STATE_HOLDOVER_OUT_OF_SPEC3: 0,
+    },
+}
+
+def is_transition_allowed(current_state, new_state):
```

Comment:
Allowed? or Expected?

Reply:
I think allowed since it's not allowed to move to this state. But I'm open to terminology change if you think it describes better